src/constants.rs

```diff
@@ -8,3 +8,9 @@ pub(crate) const NUM_FE_FOR_RO: usize = 24;
 
 /// Bit size of Nova field element hashes
 pub const NUM_HASH_BITS: usize = 250;
+
+/// The threshold of elements we check when extending a `CommitmentKey`
+/// to be convinced that they were generated from the same label
+///
+/// TODO: I have no idea about *any* security implications here.
```

Comment:
In a caching context, you should either :
- trust your local disc, in which case the checking threshold should be 0, or 1 to avoid regressions due to code changes, or
- not trust the cache, in which case the appropriate threshold is INFINITY